backend/src/api/game/play.rs

```diff
@@ -0,0 +1,49 @@
+use crate::api::game::{game_request::GameRequest, game_response::GameResponse};
+use crate::server_error::ServerError;
+use actix_web::web::{self, Json, Path};
+use hive_lib::{
+    game_error::GameError, history::History, piece::Piece, position::Position, state::State,
+};
+
+fn get_game_state_from_db(_game_id: u64) -> Result<State, GameError> {
+    let game = "game.txt";
+    let history = History::from_filepath(game)?;
+    State::new_from_history(&history)
+}
+
+fn play_turn(mut state: State, piece: String, pos: String) -> Result<GameResponse, ServerError> {
+    let board_move = format!("{} {}", piece, pos);
+    let piece = Piece::from_string(&piece)?;
+    let pos = Position::from_string(&pos, &state.board)?;
+    state.play_turn(piece, pos)?;
+    let game = "game.txt";
+    state.history.write_move(game, state.turn, board_move);
+    Ok(GameResponse::new(
```

Comment:
i think it'd be a lot cleaner if there was a `GameResponse::from_state(state: &State)`, or maybe `impl From<&State> for GameReponse`